corgi/core/models.py

```diff
@@ -848,18 +899,18 @@ class Meta:
         )
 
 
-def get_product_details(variant_names: tuple[str], stream_names: list[str]) -> dict[str, set[str]]:
+def get_product_details(
+    variant_names: tuple[str, ...], stream_names: list[str]
+) -> dict[str, set[str]]:
     """
     Given stream / variant names from the PCR table, look up all related ProductModel subclasses
 
     In other words, builds relate to variants and streams through the PCR table
     Components relate to builds, and products / versions relate to variants / streams
     We want to link all build components to their related products, versions, streams, and variants
     """
-    if variant_names:
-        variant_streams = ProductVariant.objects.filter(name__in=variant_names).values_list(
-            "productstreams__name", flat=True
-        )
+    for variant in ProductVariant.objects.filter(name__in=variant_names):
+        variant_streams = variant.get_related_names_of_type(ProductStream)
```

Comment:
We do expect that a Variant will have more than one INFERRED-type node. One Variant can get linked to lots of streams. What I changed based on your feedback from the previous PR was removing 'type' from the uniqueness constraint of ProductNode so that we can only ever have one relationship between the same obj and parent (variant and stream), so in that case (which occurs today for quarkus-2.13) we set the type to DIRECT.

We could inline this call to get_related_names_of_type. But I use it in more than one place, with a differnt mapping model in either call.